targets/bq_interface/main.cpp

```diff
@@ -0,0 +1,273 @@
+#include <stdlib.h>
+#include <string.h>
+
+#include <BMS/dev/BQ76952.hpp>
+#include <EVT/io/I2C.hpp>
+#include <EVT/io/UART.hpp>
+#include <EVT/io/manager.hpp>
+#include <EVT/io/pin.hpp>
+#include <EVT/utils/time.hpp>
+
+namespace IO = EVT::core::IO;
+namespace time = EVT::core::time;
+
+constexpr size_t MAX_BUFF = 100;
+
+char inputBuffer[MAX_BUFF];
+
+/**
+ * Function for making a direct read request
+ *
+ * @param[in] uart The UART interface to read in from
+ * @param[in] bq The BQ interface to communicate with
+ */
+void directRead(IO::UART& uart, BMS::DEV::BQ76952& bq) {
+    uart.printf("Enter the direct address in hex: 0x");
+    uart.gets(inputBuffer, MAX_BUFF);
+    uart.printf("\r\n");
+
+    uint32_t reg = strtol(inputBuffer, nullptr, 16);
+
+    uint16_t regValue = 0;
+    auto result = bq.makeDirectRead(reg, &regValue);
+
+    // Make sure the read was successful
+    if (result != BMS::DEV::BQ76952::Status::OK) {
+        uart.printf("Failed to read register: 0x%x\r\n", reg);
+        return;
+    }
+
+    // Print the result
+    uart.printf("Register 0x%x: 0x%04X\r\n", reg, regValue);
+}
+
+/**
+ * Function for making a subcommand request
+ *
+ * @param[in] uart The UART interface to read from
+ * @param[in] bq The BQ interface
+ */
+void subcommandRead(IO::UART& uart, BMS::DEV::BQ76952& bq) {
+    uart.printf("Enter the subcommand address in hex: 0x");
+    uart.gets(inputBuffer, MAX_BUFF);
+    uart.printf("\r\n");
+
+    // Determine the target subcommand register
+    uint16_t reg = strtol(inputBuffer, nullptr, 16);
+
+    // Make the read request
+    uint32_t subcommandValue = 0;
+    auto result = bq.makeSubcommandRead(reg, &subcommandValue);
+
+    // Make sure the read was successful
+    if (result != BMS::DEV::BQ76952::Status::OK) {
+        uart.printf("Failed to read register: 0x%x\r\n", reg);
+        return;
+    }
+
+    uart.printf("Register 0x%x: 0x%08X\r\n", reg, subcommandValue);
+}
+
+/**
+ * Function for making a RAM read request
+ *
+ * @param[in] uart The UART interface to read in from
+ * @param[in] bq The BQ interface to use
+ */
+void ramRead(IO::UART& uart, BMS::DEV::BQ76952& bq) {
+    uart.printf("Enter the RAM address in hex: 0x");
+    uart.gets(inputBuffer, MAX_BUFF);
+    uart.printf("\r\n");
+
+    // Determine the target subcommand register
+    uint16_t reg = strtol(inputBuffer, nullptr, 16);
+
+    // Make the read request
+    uint32_t ramValue = 0;
+    auto result = bq.makeRAMRead(reg, &ramValue);
+
+    // Make sure the read was successful
+    if (result != BMS::DEV::BQ76952::Status::OK) {
+        uart.printf("Failed to read register: 0x%x\r\n", reg);
+        return;
+    }
+
+    uart.printf("Register 0x%x: 0x%08X\r\n", reg, ramValue);
+}
+
+/**
+ * Function for making a direct write request
+ *
+ * @param[in] uart The UART interface to write in from
+ */
+void directWrite(IO::UART& uart) {}
```

Comment:
Did you mean to implement these functions?

Reply:
Not at the moment, later this may be needed for testing, but for this functionality it wasn't needed